bfabric/src/result_container.py

```diff
@@ -9,18 +17,88 @@ class BfabricResultType(Enum):
     LISTZEEP = 3
 
 
+def _clean_result(rez: dict, drop_empty: bool = True, drop_underscores_suds: bool = True,
+                  sort_responses: bool = False) -> dict:
+    if drop_empty:
+        formatter.drop_empty_elements(rez, inplace=True)
+
+    if drop_underscores_suds:
+        formatter.map_element_keys(rez,
+                                  {'_id': 'id', '_classname': 'classname', '_projectid': 'projectid'},
+                                  inplace=True)
+    if sort_responses:
+        formatter.sort_dicts_by_key(rez, inplace=True)
+
+    return rez
+
+
 class ResultContainer:
-    def __init__(self, results: list, resultType: BfabricResultType):
+    def __init__(self, results: list, result_type: BfabricResultType, total_pages_api: int = None):
+        """
+        :param results:    List of BFabric query results
+        :param result_type: Format of each result (All must be of the same format)
+        :param total_pages_api:  Maximal number of pages that were available for reading.
+            NOTE: User may have requested to cap the total number of results. Thus, it may be of interest to know
+            the (approximate) total number of results the API had for the query. The total number of results is
+            somewhere between max_pages * (BFABRIC_QUERY_LIMIT - 1) and max_pages * BFABRIC_QUERY_LIMIT
+        """
         self.results = results
-        self.resultType = resultType
+        self.result_type = result_type
+        self.total_pages_api = total_pages_api
+
+    def __getitem__(self, idx: int):
+        return self.results[idx]
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        return str(self.to_list_dict())
+
+    def __len__(self):
+        return len(self.results)
+
+    def append(self, other: ResultContainer) -> None:
```

Comment:
I like the suggestion. I really don't understand why is it offering me to commit this by clicking commit suggestion. This is not a full code refactor, and will only change the name of the function, without changing the name in the places where the function is used. This would create a non-functioning code 0_0

Reply:
It is, because I created the suggestion to illustrate the point, but you are right that committing just that would not work... I should have added `requires further changes` or something like that to be clear. :)